added functionality to install vector from repository

beechesII opened the pull request:
* added functionality to install vector from repository
* updated README.md
* moved sources,transforms and sinks from defaults to converge.yml
* updated galaxy.yml
* split main.yml into mulitple files

We tested this MR repo on CentOS7 with installation from repository. 

Resolves #6

beechesII committed:
added functionality to install vector from repository

beechesII committed:
added functionality to install vector from repository

beechesII committed:
changed baseurl for yum repository

moved sources,transforms and sinks from defaults to converge.yml
updated README.md

remove duplicate entry in readme.md

beechesII committed:
changed baseurl for yum repository

beechesII committed:
moved sources,transforms and sinks from defaults to converge.yml
updated README.md

spylik commented on README.md:
```diff
@@ -5,6 +5,64 @@
+| vector_nightly | no | false | use vector nightly build
+| add_vector_docker_group | no | false | add user vector to group docker
+| add_vector_journal_group | no | false | add user vector to group systemd-journal
+| add_vector_journal_group | no | false | add user vector to group systemd-journal
```
small typeo: `add_vector_journal_group` mentioned twice

beechesII replied:
I removed the duplicate in https://github.com/dzervas/ansible-vector/pull/17/commits/39c971980b48d37bbe44e7dd1167f760771cda52.

beechesII committed:
remove duplicate entry in readme.md

Commit:
Fix deb package url

gaetanfl committed:
Merge pull request #18 from micwiel/master

Fix deb package url

gaetan-craft committed:
add files required by ansible-galaxy

gaetanfl committed:
Merge pull request #19 from gaetan-craft/fix_ansible_galaxy

add files required by ansible-galaxy

gaetan-craft committed:
fix tests

gaetanfl commented:
Thanks for your work. I'm sorry I didn't see it before. I somehow left using vector at work for quite a long time.
I made some PRs to fix tests and release process before taking care of your PR.
Do you mind rebasing it before I review

gaetanfl committed:
Merge pull request #20 from gaetan-craft/fix_tests

Fix tests

gaetan-craft committed:
Add a template for yaml config generation and ensure vector use the config file

gaetan-craft committed:
Add test for config file usage

gaetanfl committed:
Merge pull request #21 from gaetan-craft/add_yaml_configuration

Add yaml configuration

gaetanfl committed:
Prepare release

xFuture603 committed:
rebased pull request branch before merge

xFuture603 commented:
> Thanks for your work. I'm sorry I didn't see it before. I somehow left using vector at work for quite a long time. I made some PRs to fix tests and release process before taking care of your PR. Do you mind rebasing it before I review

Hello @gaetanfl,

We've rebased our work before your review.

beechesII committed:
make repository configurable and vector version, in case user want to use a repo proxy and want to configure the installed vector version
added variables vector_repo, vector_repo_key and vector_package
defined defaults for new variables
updated README.md

beechesII committed:
fix repo creation

beechesII committed:
make transforms realy not required

beechesII committed:
Update vector repository and lint project (#2)

update vector repositories and lint project

Co-authored-by: Christopher Grau <beechesII@users.noreply.github.com>
Co-authored-by: sjedlick <104558780+sjedlick@users.noreply.github.com>

sjedlick committed:
allow configuring RestartSec in systemd config (#5)

* feat(NOISSUE): allow configuring RestartSec in systemd config
---------

Co-authored-by: Swen Jedlicka <swen.jedlicka@telekom.de>
Co-authored-by: schurzi <github@drachen-server.de>

beechesII committed:
separation of repo configuration and installation

beechesII committed:
fix ansible-lint action

beechesII committed:
fix var-naming[no-role-prefix]: Variables names from within roles should use vector_ as a prefix. (register: download_repo_key)

beechesII committed:
bump version to 2.0.0

beechesII committed:
add vector_manage_repo to molecule test case

beechesII committed:
enhancement - add molecule test for debian 13

beechesII committed:
fix instance creation

schurzi committed:
Merge pull request #7 from telekom-mms/separation_of_repo_configuration_and_installation

Separation of repository configuration and installation

schurzi committed:
Merge pull request #8 from telekom-mms/6-enhancement-add-molecule-test-for-debian-13

Add Molecule tests for Debian 13

Commit:
fix: update syntax for ansible_facts in install and repo tasks

schurzi committed:
Merge pull request #9 from sjedlick/fix/new_syntax_for_ansible_v13

Update syntax for ansible_facts in tasks to support ansible-core 2.19

beechesII committed:
fix - meta/main.yml (#10)

Co-authored-by: Christopher Grau <c.grau@telekom.de>
